libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ModelOverview.tsx

```diff
@@ -485,6 +480,65 @@ export class ModelOverview extends React.Component<
     );
   }
 
+  private shouldRenderModelOverviewChartPivot(): boolean {
+    return (
+      (!this.state.datasetCohortChartIsVisible &&
+        this.state.selectedFeatureBasedCohorts !== undefined &&
+        this.state.selectedFeatureBasedCohorts.length > 0) ||
+      (this.state.datasetCohortChartIsVisible &&
+        this.state.selectedDatasetCohorts !== undefined &&
+        this.state.selectedDatasetCohorts.length > 0)
+    );
+  }
+
+  private updateDatasetCohortStats(): void {
+    const datasetCohortMetricStats = generateMetrics(
+      this.context.jointDataset,
+      this.context.errorCohorts.map((errorCohort) =>
+        errorCohort.cohort.unwrap(JointDataset.IndexLabel)
+      ),
+      this.context.modelMetadata.modelType
+    );
+
+    this.setState({
+      datasetBasedCohorts: this.context.errorCohorts,
+      datasetCohortLabeledStatistics: datasetCohortMetricStats
+    });
+  }
+
+  private async updateFeatureCohortStats(): Promise<void> {
+    // generate table contents for selected feature cohorts
+    const featureBasedCohorts = generateOverlappingFeatureBasedCohorts(
+      this.context.baseErrorCohort,
+      this.context.jointDataset,
+      this.context.dataset,
+      this.state.selectedFeatures,
+      this.state.selectedFeaturesContinuousFeatureBins
+    );
+
+    const featureCohortMetricStats = generateMetrics(
+      this.context.jointDataset,
+      featureBasedCohorts.map((errorCohort) =>
+        errorCohort.cohort.unwrap(JointDataset.IndexLabel)
+      ),
+      this.context.modelMetadata.modelType
+    );
+
+    this.setState({
+      featureBasedCohortLabeledStatistics: featureCohortMetricStats,
+      featureBasedCohorts
+    });
+  }
+
+  private ifCohortIndexesEquals(a: number[], b: number[]): boolean {
+    return (
+      Array.isArray(a) &&
```

Comment:
Perhaps something like https://docs-lodash.com/v4/is-equal/ might be useful to avoid reinventing the wheel?